src/main/java/newbank/server/exceptions/CustomerMaxAccountsException.java

```diff
@@ -0,0 +1,13 @@
+package newbank.server.exceptions;
+
+public class CustomerMaxAccountsException extends Exception {
+  private int maxAccounts;
+
+  public CustomerMaxAccountsException(int maxAccounts) {
+    this.maxAccounts = maxAccounts;
```

Comment:
I wouldn't pass the maxAccounts parameter to the exception class. You already know it, it is a public static member of the Account class.

Kindest Regards,
Jules

Reply:
One argument for using maxAccounts like this is that it might (or might not, of course :smile:) be more natural in usage.

No matter in what place an account would be created, the same exception would be thrown and the warning would be shown. Then, the user of the constructor can obtain all the meta information from that one exception, without diving into the implementation of the (more sophisticated) class Account.

WDYT, Jules?

Best,
Arek

Reply:
I think the maxAccounts properties has been correctly declared as a property of the Account class (public static final).
It is therefore accessible to all clients that need that information. The exception class doesn't need to hold a separate copy of that I would say. It is better kept in one place only.

Kindest Regards,
Jules